### ISSUE TITLE
Make aliases useful for deprecating commands

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#3342]
The current alias system is intended to allow a player to make a shorter version of frequently used commands and/or parameter lists. However, aliases have recently begun to be used as a way to allow an old name for a renamed/deprecated tool to function. The idea is that players who are used to typing the old name will still be able to do so, and renamed tools that can be `enable`d will still be `enable`able via their old name.

Aliases, however, are not well integrated into the rest of the system and do not really provide players and script writers with a good overall experience when they are used to make deprecated names "still available". Here are some ideas for improving the player experience for aliases:

- [ ] show appropriate help in `gui/launcher` when the alias is in the commandline. It is debatable whether they should be in the autocomplete list, but at least the help text for the command that the alias expands to should be shown in the help area. It would also be useful to prepend the help text with the alias expansion so the player can see exactly what they will be running.
- [ ] possibly already covered by the `gui/launcher` item, `helpdb` should display useful help text if an alias is given as an entry. modifying aliases should poke `helpdb` to reindex
- [ ] allow `reqscript` to find the correct script if an alias is named (e.g. `reqscript('unforbid')` would redirect to `reqscript('claim')`)
- [ ] allow `require` to find the correct module if an alias is named (e.g. `require('plugins.oldname')` would redirect to `require('plugins.newname')`)

**Comment by Penumbra69** [comment]
Taking a look at this.

**Comment by myk002** [comment]
do you need any help on this one? I could do this part to unblock the claim rename PRs

**Comment by jppessoap** [comment]
Hello. Could aliases be made for the commands "show" and "hide" so they include "console" or "terminal" on their name, which would better reflect what exactly is shown or hidden, making finding those commands easier?

**Comment by Penumbra69** [comment]
> 

Apologies - I had some personal matters to deal with -- I will dig back into this. 
I believe I know what needs to be done, will try to get a PR pushed asap.

**Comment by Penumbra69** [comment]
Almost ready to push this PR -- one Q:
' document that players should poke helpdb to reindex after modifying aliases' 

What kind of documentation are you suggesting here?

**Comment by myk002** [comment]
in the `alias` command docs, just a line that says something to the effect of "help text is refreshed whenever a game is loaded, but if you modify alias definitions during a running game and want to enable autocomplete and help text for the alias in `gui/launcher`, be sure to run this command: `:lua require('helpdb').refresh()`

**Comment by Penumbra69** [comment]
> in the `alias` command docs, just a line that says something to the effect of "help text is refreshed whenever a game is loaded, but if you modify alias definitions during a running game and want to enable autocomplete and help text for the alias in `gui/launcher`, be sure to run this command: `:lua require('helpdb').refresh()`"

My implementation doesn't seem to incur this requirement, as I'm not actually changing the `helpdb` text, but synthesizing the alias text on top of what `helpdb` returns.  When I add an alias in the console, `gui/launcher` can use it without a refresh or reload since it's using the C++ functions to test the input with.  

I'll push the PR to ensure my solution meets your needs.

**Comment by myk002** [comment]
In order to have the alias appear in the autocomplete list, as well as the output of `ls` and `tags` (which I realize was not clearly stated as a requirement), I'm certain you'll have to add the entry to helpdb. It will likely only take a light refactoring of logic you've added to `gui/launcher`. I'll take a look in the PR.

**Comment by Penumbra69** [comment]
I did not recognize that we wanted the alias as well to be listed - I was only solving for showing the appropriate help when an alias was detected.  I will dig into this this evening.
